src/RwaSpell.sol

```diff
@@ -0,0 +1,235 @@
+pragma solidity 0.5.12;
+
+import "lib/dss-interfaces/src/dss/VatAbstract.sol";
+import "lib/dss-interfaces/src/dapp/DSPauseAbstract.sol";
+import "lib/dss-interfaces/src/dss/CatAbstract.sol";
+import "lib/dss-interfaces/src/dss/JugAbstract.sol";
+import "lib/dss-interfaces/src/dss/EndAbstract.sol";
+import "lib/dss-interfaces/src/dss/SpotAbstract.sol";
+import "lib/dss-interfaces/src/dss/GemJoinAbstract.sol";
+import "lib/dss-interfaces/src/dss/DaiJoinAbstract.sol";
+import "lib/dss-interfaces/src/dss/FlipAbstract.sol";
+import "lib/dss-interfaces/src/dapp/DSTokenAbstract.sol";
+import "lib/dss-interfaces/src/dss/IlkRegistryAbstract.sol";
+import "lib/dss-interfaces/src/dss/ChainlogAbstract.sol";
+import "lib/dss-interfaces/src/dapp/DSValueAbstract.sol";
+
+interface RwaLiquidationLike {
+    function wards(address) external returns (uint256);
+    function ilks(bytes32) external returns (bytes32,address,uint48,uint48);
+    function rely(address) external;
+    function deny(address) external;
+    function init(bytes32, uint256, bytes32, uint48) external;
+    function tell(bytes32) external;
+    function cure(bytes32) external;
+    function cull(bytes32) external;
+    function good(bytes32) external view;
+}
+
+interface RwaRoutingConduitLike {
+    function wards(address) external returns (uint256);
+    function can(address) external returns (uint256);
+    function rely(address) external;
+    function deny(address) external;
+    function hope(address) external;
+    function nope(address) external;
+    function bud(address) external returns (uint256);
+    function kiss(address) external;
+    function diss(address) external;
+    function pick(address) external;
+    function push() external;
+}
+
+interface RwaUrnLike {
+    function hope(address) external:
+}
+
+contract SpellAction {
+    // KOVAN ADDRESSES
+   //
+    // The contracts in this list should correspond to MCD core contracts, verify
+    // against the current release list at:
+    //     https://changelog.makerdao.com/releases/mainnet/active/contracts.json
+    ChainlogAbstract constant CHANGELOG =
+        ChainlogAbstract(0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F);
+
+    address constant MCD_VAT      = 0xbA987bDB501d131f766fEe8180Da5d81b34b69d9;
```

Comment:
Let's pull these in from the changelog.

Reply:
In that case they can't be `constant` I think. Is it worth putting these all into storage at deploy time?